pkg/registry/helm.go

```diff
@@ -0,0 +1,363 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package registry
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/ghodss/yaml"
+	"github.com/ksonnet/ksonnet/pkg/app"
+	"github.com/ksonnet/ksonnet/pkg/parts"
+	"github.com/ksonnet/ksonnet/pkg/util/archive"
+	"github.com/pkg/errors"
+)
+
+// Helm is a Helm repository.
+type Helm struct {
+	app              app.App
+	spec             *app.RegistryRefSpec
+	repositoryClient HelmRepositoryClient
+	unarchiver       archive.Unarchiver
+	httpGet          httpGet
```

Comment:
Why is this not encapsulated in the `repositoryClient`?

Reply:
Because I missed it. It should be in there.